### PR TITLE
[workloadmeta] Refactor to avoid requiring fx in "impl" dir

### PR DESCRIPTION
**Pull request by davidor**
### What does this PR do?

Refactors the workloadmeta code so that the `impl` does not depend on `fx`.
This is done to follow the new conventions of the component framework.

/cc @dustmop 

### Describe how to test/QA your changes

Skip. Should be covered by e2e tests.

### CODE REVIEW

File: comp/languagedetection/bundle.go [hunk 1]
```diff
@@ -1,26 +0,0 @@
-// Unless explicitly stated otherwise all files in this repository are licensed
```
Comment (davidor) [inline comp/languagedetection/bundle.go:1]: I'm not sure about the purpose of this file.
MockBundle() is not used anywhere, and Bundle() is only used in the test file below.
It's the only test that failed after the refactor.

File: comp/core/workloadmeta/impl/store_test.go [hunk 2]
```diff
@@ -31,19 +31,28 @@ const (
...
+	fx.In
+	Config config.Component
 }
 
-func TestHandleEvents(t *testing.T) {
-
-	deps := fxutil.Test[dependencies](t, fx.Options(
-		fx.Provide(func() log.Component { return logmock.New(t) }),
+func newWorkloadmetaObject(t *testing.T) *workloadmeta {
+	testDeps := fxutil.Test[testDependencies](t, fx.Options(
```
Comment (ogaca-dd) [inline comp/core/workloadmeta/impl/store_test.go:40]: As you are still use `fxutil.Test`, `fx` is still used in `impl` folder. 
I am not sure removing `fxutil.Test` is a great idea for now, so you can probably keep it.

**Review by ogaca-dd — APPROVED** [review]

File: .github/CODEOWNERS [hunk 3]
```diff
@@ -280,6 +279,7 @@
 /comp/trace/etwtracer @DataDog/windows-agent
 /comp/autoscaling/datadogclient @DataDog/container-integrations
 /comp/etw @DataDog/windows-agent
+/comp/languagedetection/client @DataDog/container-platform
```
Comment (davidor) [inline .github/CODEOWNERS:282]: This change is needed by the linter. It's introduced by `inv components.lint-components --fix`.

File: comp/core/workloadmeta/impl/store_test.go (hunk 2 above)
Comment (davidor) [reply comp/core/workloadmeta/impl/store_test.go:40]: Indeed. I noticed it but thought it was fine in test files.
Any suggestion about how to remove this `fx` dependency here? It's only used in the line that you mentioned and the `testDependencies` struct defined above.

Comment (ogaca-dd) [reply comp/core/workloadmeta/impl/store_test.go:40]: I think you can keep your code as it is.

**Review by clamoriniere — APPROVED** [review]

**Comment by davidor** [comment]
Rebased on top of main to fix a conflict.

**Pull request merged**